### PR TITLE
Object notifications

### PR DESCRIPTION
**Pull request by blagoev**
Suport RealmObject change notifications #105 
Supersedes #207 

Needs generator changes tracked in PR 269

### CODE REVIEW

File: src/realm_dart_collections.cpp [hunk 1]
```diff
@@ -89,12 +109,13 @@ RLM_API realm_notification_token_t* realm_dart_results_add_notification_callback
...
-    CallbackData* callback_data = new CallbackData(notification_controller, callback);
+    auto callback_data = new CollectionCallbackData(notification_controller, callback);
 
     return realm_results_add_notification_callback(results,
         callback_data,
-        free_callback,
-        on_change_callback,
+        free_collection_callback_data,
+        on_collection_change_callback,
+        // on_collection_change_callback_type<CollectionCallbackData>,
```
Comment (nielsenko) [inline src/realm_dart_collections.cpp:118]:
```suggestion
```

File: test/realm_test.dart [hunk 2]
```diff
@@ -1002,7 +1003,99 @@ Future<void> main([List<String>? args]) async {
...
+        var realm = Realm(config);
+
+        final team = Team('t1', players: [Person("p1")]);
+        realm.write(() => realm.add(team));
+
+        var firstCall = true;
+        final subscription = (team.players as RealmList<Person>).changes.listen((changes) {
+          if (firstCall) {
+            firstCall = false;
+            expect(changes.inserted.isEmpty, true);
```
Comment (nielsenko) [inline test/realm_test.dart:1021]: Not really related to this PR, but I got to admit I like `insertions`, etc. better. But I guess it is in the eye of the beholder.

Comment (blagoev) [reply test/realm_test.dart:1021]: we use `changes` since it's something going to happen and `inserted` since it already happened when the event is called

Comment (nielsenko) [reply test/realm_test.dart:1021]: English is not my native tongue, but this doesn't sound right to me. To me `changes` is a description of what changed, and that may include `insertions` etc. Perhaps you could run it by a true native? Anyway, I have already approved the PR. I just noticed it differs from core and .NET, and I felt I preferred their wording.

Comment (blagoev) [reply test/realm_test.dart:1021]: .NET uses `inserted`, `modified` etc.
Anyways I already added an agenda topic to discuss this.

**Pull request merged**